x-pack/plugins/security_solution/public/timelines/components/timeline/event_details.tsx

```diff
@@ -11,39 +11,105 @@
  */
 
 import { EuiSpacer } from '@elastic/eui';
-import React from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import deepEqual from 'fast-deep-equal';
+import { find } from 'lodash/fp';
 
+import { useDispatch } from 'react-redux';
 import { BrowserFields, DocValueFields } from '../../../common/containers/source';
 import {
   ExpandableEvent,
   ExpandableEventTitle,
+  OnEventDetailsClose,
 } from '../../../timelines/components/timeline/expandable_event';
 import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { TimelineEventsDetailsItem } from '../../../../common/search_strategy/timeline';
+import { useTimelineEventsDetails } from '../../containers/details';
+import { TimelineId } from '../../../../common/types/timeline';
+import { activeTimeline } from '../../containers/active_timeline_context';
+import { timelineActions } from '../../store/timeline';
 
 interface EventDetailsProps {
   browserFields: BrowserFields;
   docValueFields: DocValueFields[];
   timelineId: string;
+  onEventDetailsClose?: OnEventDetailsClose;
 }
 
 const EventDetailsComponent: React.FC<EventDetailsProps> = ({
   browserFields,
   docValueFields,
   timelineId,
+  onEventDetailsClose,
 }) => {
-  const expandedEvent = useDeepEqualSelector(
-    (state) => state.timeline.timelineById[timelineId]?.expandedEvent
+  const dispatch = useDispatch();
+
+  const { expandedEvent, isSaving } = useDeepEqualSelector(
+    (state) => state.timeline.timelineById[timelineId]
   );
 
+  const [loading, detailsData] = useTimelineEventsDetails({
+    docValueFields,
+    indexName: expandedEvent.indexName!,
+    eventId: expandedEvent.eventId!,
+    skip: !expandedEvent.eventId,
+  });
+
+  const isAlert = useMemo(() => {
+    if (detailsData) {
+      const signalField = find({ category: 'signal', field: 'signal.rule.id' }, detailsData) as
+        | TimelineEventsDetailsItem
+        | undefined;
+
+      if (signalField?.originalValue) {
+        return true;
+      }
+    }
+    return false;
+  }, [detailsData]);
+
+  const handleOnEventClosed = useCallback(() => {
+    dispatch(
+      timelineActions.toggleExpandedEvent({
+        timelineId,
+        event: null,
+      })
+    );
+
+    if (timelineId === TimelineId.active) {
+      activeTimeline.toggleExpandedEvent({
+        eventId: expandedEvent.eventId,
+        indexName: '',
+        loading: false,
+      });
+    }
+
+    if (onEventDetailsClose) {
+      onEventDetailsClose();
+    }
+  }, [dispatch, timelineId, expandedEvent.eventId, onEventDetailsClose]);
+
+  useEffect(() => {
+    if (isSaving) {
+      handleOnEventClosed();
```

Comment:
why we want to hide the details when saving a timeline?

Reply:
As it is expected to clear the expanded event after we've done a filter or search in https://github.com/elastic/kibana/pull/83963#issuecomment-736405437 (last two items).

Reply:
I think `isSaving` is going to be toggled when editing a title/description/toggling favorite as well and I don't expect the details to be closed in that case

Reply:
I'd move this logic to `events_viewer` and `query_tab_content/index` and add `useEffect` to listen on the response of `useTimelineEvents` and check if the `expandedEvent` exists on the `data`

Reply:
ok, I've put the logic into query_tab_content/index. In regarding to events_viewer, as the expand event's button is disabled after clicked, I updated the tooltip to always show `Expand event`, therefore  I am not applying the functionality of `closeEventDetails` to it;